test/spec/videoCache_spec.js

```diff
@@ -297,6 +298,46 @@ describe('The video cache', function () {
       JSON.parse(request.requestBody).should.deep.equal(payload);
     });
 
+    it('should wait the duration of the batchTimeout if batched requests are enabled in the config', () => {
+      config.setConfig({
+        cache: {
+          url: 'https://prebid.adnxs.com/pbc/v1/cache',
+          batchSize: 3,
+          batchTimeout: 20
+        }
+      });
+
+      let mockTimeout = sinon.stub().callsFake((cb) => cb());
+      let stubCache = sinon.stub();
+      const batchAndStore = batchingCache(mockTimeout, stubCache);
+      batchAndStore();
+      sinon.assert.calledWith(mockTimeout, sinon.match.any, 20);
+    });
+
+    // it('should batch requests correctly based on the value of batchSize if batched requests are enabled in the config', () => {
```

Comment:
Had one question here.  I can almost get this test to work.  On line 330 below I call `batchAndStore` 3 times to simulate 3 bidResponses coming in and then want to test that all 3 are put into a batchSize of 3.  However, on line 571 in auction.js, the setTimeout gets ignored during this test and line 338 below is failing:

expected:
[
  { afterBidAdded: undefined, auctionInstance: undefined, bidResponse: undefined },
  { afterBidAdded: undefined, auctionInstance: undefined, bidResponse: undefined },
  { afterBidAdded: undefined, auctionInstance: undefined, bidResponse: undefined }
]

actual:
call 1
[
  { afterBidAdded: undefined, auctionInstance: undefined, bidResponse: undefined }
]

call 2
[
  { afterBidAdded: undefined, auctionInstance: undefined, bidResponse: undefined }
]

call 3
[
  { afterBidAdded: undefined, auctionInstance: undefined, bidResponse: undefined }
]

Is there a way to get past this setTimeout issue? (or perhaps i should just re-write my test below in a different way).  I tried using fake timers in sinon but didn't have any luck

Reply:
Thanks!  Your suggestion totally makes sense.  I updated the test, and now it's almost 100% working, but have one last question.  I can see that sinon now expects the correct batch, yet the `expectedBatch` var i created seems to get formatted while i run my test and fails.  i tried using `// eslint-disable-next-line` but that didn't seem to have an effect

Reply:
can see the output in the circle ci test that failed